skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
```

Comment:
Sounds good. So the GLCM will be len(levels) x len(levels), if levels is array like, and [0, levels - 1] if levels is a positive integer. This I could do. 

However,  `levels` cannot default to `array.ravel().unique()`, as this would result in different matrices for different images. For most applications, this is not the desired behavior. Also, it would change the default behavior, which is probably also not what we want.